feat(corrections-modal): Improve Corrections Modal UX

Takshil-Kunadia opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/1/26890605006346/project/1209292256643614/task/1209683493118756?focus=true.

- Moved away from the Accordion to a simpler Card Layout.
- Add Notices for visual confirmation of action success.

### How to test the changes in this Pull Request:

1. Open a Post > Corrections & Clarification section in Sidebar > Manage Correction.
2. Click on Add New Correction > Add details > Click Add.
3. Observe Notice being shown above the correction being added. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Screenshots:-

https://github.com/user-attachments/assets/ff7ed691-cf70-42a3-9d64-e9f26134b5a3

Takshil-Kunadia committed:
feat(corrections-modal): enhance corrections modal UX with notices and card layout

Takshil-Kunadia commented:
Waiting for https://github.com/Automattic/newspack-plugin/pull/3829 to be merged.

thomasguillot commented:
https://github.com/user-attachments/assets/64b463c4-9554-467a-b588-6c6dca01f72b

Something we noticed is when adding an additional correction/clarification, the behaviour is a bit confusing — there’s no success message nor does the log automatically open to display the newest addition. You have to manually open the log to make sure it got added.

Ideally we need an isBusy on the “Add” button, and a Snackbar to let the user know it’s been successfully added to the Corrections log.

Takshil-Kunadia commented:
Thanks! @thomasguillot I have implemented precisely these same change via this PR. Please refer to the Demo video in the PR description.

thomasguillot commented:
@Takshil-Kunadia Can we use the `Snackbar` component instead of the `Notice`?

Takshil-Kunadia committed:
feat(corrections-modal): replace success notice with snackbar

Takshil-Kunadia commented:
@thomasguillot replaced notice with snackbar ✅ . The snackbar disappears after 10 seconds. Also added an add more button so user's don't have to scroll down manually to add new corrections. 

https://github.com/user-attachments/assets/f2dd5817-d81d-436a-af0b-d9598aeaa336

thomasguillot commented:
I meant a snackbar in the bottom left corner, just like when we save a post for example.

Takshil-Kunadia commented:
@thomasguillot The default snackbar has a lower z-index than a modal component. It would appear below the modal overlay.

https://github.com/user-attachments/assets/be4161ae-4b7a-4321-998c-1b31f0acfb9e

thomasguillot commented:
> The default snackbar has a lower z-index than a modal component. It would appear below the modal overlay.

Can't we increase the z-index for this specfic snackbar using css?

Takshil-Kunadia commented:
> Can't we increase the z-index for this specfic snackbar using css?

I couldn't find a way to add an identifier or className to the notice, alternatively I adjusted the modal's z-index instead to resolve the issue.

https://github.com/user-attachments/assets/4b4d34a6-57b0-4968-9af5-09b2f79529ac

Takshil-Kunadia committed:
refactor(corrections-modal): replaced contained notice with global notice

thomasguillot commented:
Thank you!

thomasguillot committed:
feat(corrections-modal): reorganise panels

thomasguillot commented:
@Takshil-Kunadia I pushed a few tweaks to the modal: https://github.com/Automattic/newspack-plugin/pull/3835/commits/e816945cc7b0f316ecb1ad9d1722b804909cef65

✅ Design

Takshil-Kunadia commented:
Nice touch! Really liked the delete notification 💡 . Thanks! @thomasguillot

leogermani commented:
This is looking so much better! But I'm afraid there's still one thing we need to improve.

When we add a new correction, we see the snackbar and we might have the wrong impression that the correction is saved. If we the Save & Close button is not visible, the user can just close the modal and the correction won't ever be saved...

Ideas:
* Make the save button be always saved
* Alert if they try to close the modal with unsaved changes
* Save changes also on post update...

I think the first option, along with removing the X button to close the moda, can be the easier?

Takshil-Kunadia committed:
feat(corrections-modal): enforce correction saving

Takshil-Kunadia commented:
Nice catch! @leogermani 

I have removed the close button, closing on escape or clicking outside, to enforce save button be clicked(or cancel) before the modal closes. Also added a separate changes have been saved notice after saving.


https://github.com/user-attachments/assets/3bcacfd9-9609-4da3-8f19-8f14df1c2773

leogermani commented:
@Takshil-Kunadia I'm approving this PR but there are 2 things I think we can improve. We can do it in follow up PRs, no rush

1. Make the bottom section with the buttons fixed in the modal, so they are always visible
2. If you add a correction and click "Cancel", the modal closes without saving. If you open the modal again, the unsaved correction will still be there, which can be confusing

Both are low priority. I'm ok adding to the backlog and working on them later
